src/config.rs

```diff
@@ -441,18 +445,25 @@ impl fmt::Debug for SharedConfigValues {
             .field("no-audio-cache", &self.no_audio_cache)
             .field("backend", &self.backend)
             .field("volume_controller", &self.volume_controller)
-            .field("device", &self.device)
-            .field("control", &self.control)
-            .field("mixer", &self.mixer)
             .field("device_name", &self.device_name)
             .field("bitrate", &self.bitrate)
             .field("initial_volume", &self.initial_volume)
             .field("volume_normalisation", &self.volume_normalisation)
             .field("normalisation_pregain", &self.normalisation_pregain)
             .field("zeroconf_port", &self.zeroconf_port)
             .field("proxy", &self.proxy)
-            .field("device_type", &self.device_type)
-            .finish()
+            .field("device_type", &self.device_type);
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "alsa_backend")] {
+                deb_struct
+                .field("device", &self.device)
+                .field("control", &self.control)
+                .field("mixer", &self.mixer)
+                .finish()
+            } else {
+                deb_struct.finish()
+            }
+        }
```

Comment:
By using the below variant, we could get rid of the `cfg_if` crate and the effect should be the same.

```suggestion
        #[cfg(feature = "alsa_backend")]
        {
            deb_struct
                .field("device", &self.device)
                .field("control", &self.control)
                .field("mixer", &self.mixer);
        }
        deb_struct.finish()
```